async_simple/coro/ConditionVariable.h

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2022, Alibaba Group Holding Limited;
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef ASYNC_SIMPLE_CORO_CONDITION_VARIABLE_H
+#define ASYNC_SIMPLE_CORO_CONDITION_VARIABLE_H
+
+#include <async_simple/coro/Lazy.h>
+
+namespace async_simple {
+namespace coro {
+
+template <class Lock>
+class ConditionVariableAwaiter;
+
+template <class Lock = void>
```

Comment:
We should handle the `Lock = void` in case of `ConditionVariable<>`. Let's remove the default argument and add a static_assert.

Reply:
We already specialized the void type，its ok to remove `Lock = void`, and update to `using Notifier = ConditionVariable<void>;`